src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let clone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        clone = {};
+        break;
+
+      case 'addProperties':
+        clone = Object.assign(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const prop of action.keysToRemove) {
+          delete clone[prop];
+        };
+        break;
+
+      default:
+        // eslint-disable-next-line no-console
+        console.log('Action not found');
```

Comment:
When none of the values fit, you just need to move on through the loop instead of outputting anything to the console, fix that